pkg/pidfile/pidfile.go

```diff
@@ -0,0 +1,50 @@
+package pidfile
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strconv"
+)
+
+type PidFile struct {
+	path string
+}
+
+func checkPidFileAlreadyExists(path string) error {
+	if pidString, err := ioutil.ReadFile(path); err == nil {
+		if pid, err := strconv.Atoi(string(pidString)); err == nil {
+			if _, err := os.FindProcess(pid); err == nil {
```

Comment:
See this issue:
https://codereview.appspot.com/7392048/
and https://groups.google.com/forum/#!topic/golang-nuts/hqrp0UHBK9k